src/Modal/Modal.jsx

```diff
@@ -5,19 +5,55 @@ require("!style!css!less!./Modal.less");
 var DEFAULT_WIDTH = 400;
 
 export var Modal = React.createClass({
+  getDefaultProps: function() {
+    return {
+      width: DEFAULT_WIDTH
+    };
+  },
+
+  getInitialState: function() {
+    return {
+      windowHeight: window.innerHeight
+    };
+  },
+
+  handleKeyUp: function(e) {
+    if (e.keyCode == 27){
+      this.props.closeModal();
+    }
+  },
+
+  handleResize: function(e) {
+    this.setState({windowHeight:window.innerHeight})
+  },
+
+
+  componentDidMount: function() {
+    window.addEventListener('keyup', this.handleKeyUp)
+    window.addEventListener('resize', this.handleResize)
+  },
+
+  componentWillUnmount: function() {
+    window.removeEventListener('keyup', this.handleKeyUp);
+    window.addEventListener('resize', this.handleResize)
+  },
+
   render: function() {
-    var width = this.props.width || DEFAULT_WIDTH;
+    var style = {
+      width: this.props.width + "px",
+      marginLeft: "-" + (this.props.width / 2) + "px",
+    }
+    // The content is 80% of the window height (same as the max height of the modal) less 75px (the height of the header and padding)
+    var contentHeight = this.state.windowHeight*.8-75
     return (
       <div className="Modal">
         <div className="Modal--background" onClick={this.props.closeModal}/>
-        <div
-          className="Modal--window"
-          style={{
-            width: width + "px",
-            marginLeft: "-" + (width / 2) + "px"
-          }}>
-          <div className="Modal--window--container">
+        <div ref="modalWindow" className="Modal--window" style={style} >
+          <div className="Modal--header">
+            <a className="Modal--close" onClick={this.props.closeModal}></a>
```

Comment:
This isn't accessible because `<a>` can't receive focus unless it has `href`. I think you can add `href=""` and it will work. Or just use `button`